chia/full_node/coin_store.py

```diff
@@ -99,6 +96,9 @@ async def new_block(self, block: FullBlock, additions: List[Coin], removals: Lis
             )
             await self._add_coin_record(reward_coin_r)
 
+        for coin_name in removals:
```

Comment:
generally I think it's good to add a comment about the order mattering here. given that someone made this mistake once, we know it can happen. so we should try to prevent it happening again

Reply:
I don't see a test covering this, is there one?

Reply:
Yes there is a test in TestBlockBodyValidation